packages/rocketchat-livechat/imports/server/rest/messages.js

```diff
@@ -0,0 +1,64 @@
+import LivechatVisitors from '../../../server/models/LivechatVisitors';
+
+RocketChat.API.v1.addRoute('livechat/messages', {
+	post() {
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-livechat-manager')) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		if (!this.bodyParams.visitor) {
+			return RocketChat.API.v1.failure('Body param "visitor" is required');
+		}
+		if (!this.bodyParams.visitor.token) {
```

Comment:
`this.bodyParams.visitor.token` doesn't work. You should use camelCased in the field names, such as `this.bodyParams.visitorToken`.